src/entity/scene.cpp

```diff
@@ -0,0 +1,64 @@
+#include <entity/scene.h>
+#include <assert.h>
+
+Scene::Scene() {
+    this->lights = std::map<int, Light>();
+    this->lightsCurrentIndex = this->lights.size() - 1;
+    this->environmentColor = Color(0, 0, 0);
+    this->objects = std::map<int, Box>();
+    this->objectsCurrentIndex = this->objects.size() - 1;
```

Comment:
this->lights.size()
```suggestion
    this->lightsCurrentIndex = this->lights.size();
    this->environmentColor = Color(0, 0, 0);
    this->objects = std::map<int, Box>();
    this->objectsCurrentIndex = this->objects.size();
```

Reply:
@kinhosz feito! Precisei alterar também a lógica utilizada no `addLIght` e no `addObject`, dá uma olhada se tá tudo ok.